src/run/run.c

```diff
@@ -1020,31 +1024,27 @@ static int start_transient_service(
                         pty_forward_set_handler(c.forward, pty_forward_handler, &c);
                 }
 
-                if (arg_wait) {
-                        _cleanup_free_ char *path = NULL;
-                        const char *mt;
 
-                        path = unit_dbus_path_from_name(service);
-                        if (!path)
-                                return log_oom();
+                path = unit_dbus_path_from_name(service);
+                if (!path)
+                        return log_oom();
 
-                        mt = strjoina("type='signal',"
-                                      "sender='org.freedesktop.systemd1',"
-                                      "path='", path, "',"
-                                      "interface='org.freedesktop.DBus.Properties',"
-                                      "member='PropertiesChanged'");
-                        r = sd_bus_add_match(bus, &c.match, mt, on_properties_changed, &c);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to add properties changed signal.");
+                mt = strjoina("type='signal',"
+                              "sender='org.freedesktop.systemd1',"
+                              "path='", path, "',"
+                              "interface='org.freedesktop.DBus.Properties',"
+                              "member='PropertiesChanged'");
```

Comment:
This will  now unconditionally do the `PropertiesChanged` match on D-Bus, as the `if (arg_wait) {` was dropped. Will this work on the private bus, if neither `--wait` nor `--pty` are specified? Is it necessary in these cases? Naïvely it appears like extending the `if` condition by `|| arg_pty` would be more straightforward?